### DISCUSSION TITLE
How can I make eza default in preview-tui?

### DISCUSSION THREAD

**Discussion by nemonein** [jarun/nnn#1874] [Q&A] [answered]
Hi.
The plugin preview-tui chooses `tree` as the first option.
I'd like to use `exa(eza)` over `tree`, but I don't see any option for that.
Do I have to edit the source(plugins/preview-tui) directly?
 -- removing tree related lines?? OR move up eza over tree?

**Comment by luukvbaal** [comment, marked as answer]
Not possible it seems, we only have `USE_ICONLOOKUP` for directory preview. Should probably merge that into something like `NNN_PREVIEWDIRPROG`, anologous to `NNN_PREVIEWIMGPROG`. For now your only option is to edit the plugin yes.

**Reply by jarun** [reply]
Plugins are samples, supposed to be edited and adapted by users as per their needs.

**Reply by nemonein** [reply]
Thank you for your replies.
I hope this should be documented somewhere.
--> Plugins are samples, so users have to modify the plugins to get a specific result.

**Reply by JohannesRgnr** [reply]
I use eza with preview-tui, works well. This is what I modified, starting line 342. Modify line 352 according to what you'd like to be displayed.

```
    # Otherwise, falling back to the defaults.
    if [ -d "$1" ]; then
        cd "$1" || return
        if [ "$NNN_ICONLOOKUP" -ne 0 ] && [ -f "$(dirname "$0")"/.iconlookup ]; then
            [ "$NNN_SPLIT" = v ] && BSTR="\n"
            # shellcheck disable=SC2012
            ls -F --group-directories-first | head -n "$((lines - 3))" | "$(dirname "$0")"/.iconlookup -l "$cols" -B "$BSTR" -b " "
        elif exists tree; then
            fifo_pager tree --filelimit "$(find . -maxdepth 1 | wc -l)" -L 3 -C -F --dirsfirst --noreport
        elif exists eza; then
            eza -l --group-directories-first --colour=always --no-permissions --no-user   
        else
            fifo_pager ls -F --group-directories-first --color=always
        fi
        cd ..
    elif [ "${encoding#*)}" = "binary" ]; then
        handle_mime "$1" "$mimetype" "$ext" "bin"
    else
        handle_mime "$1" "$mimetype" "$ext"
    fi
}
```
